### DISCUSSION TITLE
Typescript error: Generic type 'ɵɵDirectiveDeclaration' requires between 6 and 8 type arguments.

### DISCUSSION THREAD

**Discussion by davidtaylor-omf** [uNmAnNeR/imaskjs#969] [Q&A]
Trying to use Imask in an Angular v14 application and its reporting the following error
Error: node_modules/angular-imask/imask.directive.d.ts:49:18 - error TS2707: Generic type 'ɵɵDirectiveDeclaration' requires between 6 and 8 type arguments.

49     static ɵdir: i0.ɵɵDirectiveDeclaration<IMaskDirective<any, any, any>, "[imask]", ["imask"], { "imask": { "alias": "imask"; "required": false; }; "unmask": { "alias": "unmask"; "required": false; }; "imaskElement": { "alias": "imaskElement"; "required": false; }; }, { "accept": "accept"; "complete": "complete"; }, never, never, true, never>;

**Comment by uNmAnNeR** [comment]
@davidtaylor-omf yep. Angular major versions are not compatible. So i have to choose which one to support. The latest is chosen so the latest imask version works well with ng16. I can't support multiple builds for each angular by my own.
